### DISCUSSION TITLE
How do I access/return the formattedvalues array?

### DISCUSSION THREAD

**Discussion by smartinsight-at** [AlexaCRM/dynamics-webapi-toolkit#59] [Q&A] [answered]
When I run the toolkit against a contact I get an array consisting of Attribute and Formattedvalues, by using $contact['dmc_source'] i get the value from Attribute. I have tried $contact['FormattedValues']['dmc_source'] but that does not work?

This is the start of the code calling the contact from the result.

foreach ( $result->Entities as $contact ) {

echo $contact['dmc_source']
// returns 10000000

}

This is the data that get returned when I print_r the $contact array 

AlexaCRM\Xrm\Entity Object
(
    [Attributes] => Array
        (
            [dmc_source] => 100000000
        )

    [FormattedValues] => Array
        (
            [dmc_source] => Data Mining
        )

    [attributeState:protected] => AlexaCRM\Xrm\AttributeState Object
        (
        )

    [Id] => 3bd3f4ab-da5e-eb11-a812-xxxxxxxxx
    [LogicalName] => contact
    [KeyAttributes] => 
)

Suggestions would be welcome as I must be missing something obvious.

**Comment by georged** [comment, marked as answer]
@smartinsight-at 
`FormattedValues` as you can see from the dump is a property of the entity object. Try `$contact->FormattedValues['dmc_source']`

Thanks
